examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
+    std::shared_ptr<Opm::BlackoilPropsAdFromDeck> distributed_props=new_props;
+    Dune::CpGrid distributed_grid = *grid;
+    // At this point all properties and state variables are correctly initialized
+    // If there are more than one processors involved, we now repartition the grid
+    // and initilialize new properties and states for it.
+    if( grid->comm().size()>=1 )
+    {
+        if(!param.getDefault("output_vtk", true))
+        {
+            OPM_THROW(std::logic_error, "We only support vtk output during parallel runs");
+        }
+        grid->loadBalance();
```

Comment:
I am not sure what happens in this and the next few statements (lines 222-226):
 - After loadBalance(), is not distributed_grid already changed since it is a shallow copy (I think)?
 - Why use separate objects for grid and global_grid?
 - What do the switchTo...() methods do, given that the grids are shallow copies?